<?xml version="1.0"?>

<launch>
  <!-- 激光TF校正节点 -->
  <!-- <node pkg="vel_pkg" type="laser_tf_corrector.py" name="laser_tf_corrector" output="screen"/> -->
  
  <arg name="tf_map_scanmatch_transform_frame_name" default="scanmatcher_frame"/>
  <arg name="base_frame" default="base_footprint"/>
  <arg name="odom_frame" default="odom"/>
  <arg name="scan_subscriber_queue_size" default="5"/>
  <arg name="scan_topic" default="scan"/>
  <arg name="map_size" default="8192"/>
  <arg name="use_sim_time" default="false"/>
  
  <!-- Hector Mapping 节点 -->
  <node pkg="hector_mapping" type="hector_mapping" name="hector_mapping" output="screen">
    <!-- Frame names -->
    <param name="map_frame" value="map" />
    <param name="base_frame" value="$(arg base_frame)" />
    <param name="odom_frame" value="$(arg odom_frame)" />
    
    <!-- Tf use -->
    <param name="use_tf_scan_transformation" value="true"/>
    <param name="use_tf_pose_start_estimate" value="false"/>
    <param name="pub_map_odom_transform" value="true"/>  <!-- 改为true，让hector发布map->odom -->
    <param name="tf_timeout" value="2.0"/>

    <!-- 发布 Odometry 消息 -->
    <param name="pub_odometry" value="true"/>
    
    
    <!-- Map 参数 -->
    <param name="map_resolution" value="0.050"/>
    <param name="map_size" value="$(arg map_size)"/>
    <param name="map_start_x" value="0.5"/>
    <param name="map_start_y" value="0.5" />
    <param name="map_multi_res_levels" value="2" />
    
    <!-- Map update parameters -->
    <param name="update_factor_free" value="0.4"/>
    <param name="update_factor_occupied" value="0.9" />    
    <param name="map_update_distance_thresh" value="0.4"/>
    <param name="map_update_angle_thresh" value="0.06" />
    <param name="laser_z_min_value" value = "-1.0" />
    <param name="laser_z_max_value" value = "1.0" />
    
    <!-- Advertising config --> 
    <param name="advertise_map_service" value="true"/>
    
    <param name="scan_subscriber_queue_size" value="$(arg scan_subscriber_queue_size)"/>
    <param name="scan_topic" value="$(arg scan_topic)"/>
    
    <param name="tf_map_scanmatch_transform_frame_name" value="$(arg tf_map_scanmatch_transform_frame_name)" />
    
    <!-- 将 scanmatch_odom 重映射为 odom -->
    <!-- <remap from="scanmatch_odom" to="odom"/> -->
  </node>
  
  <!-- 使用odometry_to_tf将/odom话题转换为TF变换 -->
  <!-- <node name="odom_to_tf" pkg="vel_pkg" type="odom_to_tf.py" output="screen">
    <remap from="odom" to="odom" />
    <param name="frame_id" value="odom" />
    <param name="child_frame_id" value="base_footprint" />
  </node> -->

</launch>